rust-runtime/aws-smithy-http-server-python/src/server.rs

```diff
@@ -460,4 +455,40 @@ event_loop.add_signal_handler(signal.SIGINT,
         self.block_on_rust_signals();
         Ok(())
     }
+
+    /// Lambda main entrypoint: start the server on Lambda.
+    ///
+    /// Unlike the `run_server`, `run_lambda_server` does not spawns other processes,
+    /// it starts the Lambda server on the current process.
+    fn run_lambda_server(&mut self, py: Python) -> PyResult<()> {
+        let event_loop = self.configure_python_event_loop(py)?;
+        let app = self.build_and_configure_router(py, event_loop)?;
+        let rt = runtime::Builder::new_multi_thread()
+            .enable_all()
+            .build()
+            .expect("Unable to start a new Tokio runtime for this process");
+        rt.block_on(async move {
+            let handler = LambdaHandler::new(app);
+            let lambda = lambda_http::run(handler);
+            tracing::debug!("Starting Lambda server");
+            if let Err(err) = lambda.await {
+                tracing::error!("server error: {}", err);
+            }
+        });
+        Ok(())
+    }
```

Comment:
It's not really a server :)

```suggestion
    fn run_lambda_handler(&mut self, py: Python) -> PyResult<()> {
        let event_loop = self.configure_python_event_loop(py)?;
        let app = self.build_and_configure_router(py, event_loop)?;
        let rt = runtime::Builder::new_multi_thread()
            .enable_all()
            .build()
            .expect("Unable to start a new Tokio runtime for this process");
        rt.block_on(async move {
            let handler = LambdaHandler::new(app);
            let lambda = lambda_http::run(handler);
            tracing::debug!("Starting Lambda handler");
            if let Err(err) = lambda.await {
                tracing::error!("server error: {}", err);
            }
        });
        Ok(())
    }
```